src/adm/tiles.rs

```diff
@@ -248,11 +249,9 @@ pub async fn get_tiles(
         metrics.incr_with_tags("filter.adm.empty_response", Some(tags));
     }
 
-    let filtered: Vec<Tile> = response
-        .tiles
-        .into_iter()
-        .filter_map(|tile| {
-            state.filter.read().unwrap().filter_and_process(
+    let filtered: Vec<Tile> = stream::iter(response.tiles)
+        .filter_map(|tile| async {
```

Comment:
That snippet's missing the original `take/collect`, re-adding those should get rid of the borrows.

You can also nest a new block instead of calling `drop`, but either way works.

e.g. 
```
let filtered = {
    let filter = state.filter.read().await;
     response.tiles...
     .collect()
}
```